workflows/kallisto-quant/processing-kallisto.Rmd

```diff
@@ -0,0 +1,150 @@
+---
+title: "Kallisto bustools Processing"
+output: html_notebook
+---
+
+This notebook is an example of processing a kallsto|bustools run as implemented in the `run-kallisto.nf` workflow.
+
+```{r setup}
+library(magrittr)
+library(ggplot2)
+library(SingleCellExperiment)
+```
+
+
+First we will download a data set to work with from S3. 
+This set was quantified as part of a benchmarking set.
+This step assumes that `awscli` is installed (which it is if you use the `scpca-r` docker image) and that you have already set up credentials.
+
+```{r}
+data_dir <- file.path("data", "kallisto", "834")
+dir.create(data_dir, recursive = TRUE, showWarnings = FALSE)
+quant_s3 <- "s3://nextflow-ccdl-results/scpca-benchmark/quants/kallisto/834-kallisto_cdna"
+mito_s3 <- "s3://nextflow-ccdl-data/reference/homo_sapiens/ensembl-100/annotation/Homo_sapiens.ensembl.100.mitogenes.txt" 
+mito_file <- file.path("data", "Homo_sapiens.ensembl.100.mitogenes.txt")
+```
+
+### Sync S3 files
+
+```{r}
+sync_call <- paste("aws s3 sync", quant_s3, data_dir, "--exclude 'bus/*'")
+system(sync_call, ignore.stdout = TRUE)
+
+# get the mitochondrial gene set
+cp_call <- paste("aws s3 cp", mito_s3, mito_file)
+system(cp_call)
+```
+
+## Read counts matrix
+```{r}
+file_base <- file.path(data_dir, "counts", "gene_count")
+
+counts <- Matrix::readMM(paste0(file_base, ".mtx")) %>%
+  # compress and transpose to genes in rows x cells in columns
+  t() %>%
+  as("dgCMatrix")
+# add annotation info
+dimnames(counts) <- list(readLines(paste0(file_base, ".genes.txt")),
+                         readLines(paste0(file_base, ".barcodes.txt")))
+```
+
+## Identifying cells with droplets
+
+### Cellranger default method first.
+
+Based on library size, capturing cells with at least 10% of the UMI counts of the upper 99th percentile.
+
+```{r}
+system.time(drops <- DropletUtils::defaultDrops(counts))
+
+sum(drops)
+min(colSums(counts[,drops]))
+```
+
+### Knee plot method
+
+Find the knee, keep the cells above that rank.
+
+```{r}
+system.time(rank_df <- DropletUtils::barcodeRanks(counts))
+knee = metadata(rank_df)$knee
+inflection = metadata(rank_df)$inflection
```

Comment:
```suggestion
system.time(rank_df <- DropletUtils::barcodeRanks(counts))
knee <- metadata(rank_df)$knee
inflection <- metadata(rank_df)$inflection
```

Reply:
Is there a reason `=` needs to be used for assignment here? I didn't think that deeply about it; I had a knee jerk reaction.